components/script/dom/messageevent.rs

```diff
@@ -219,7 +238,13 @@ impl MessageEventMethods for MessageEvent {
 
     /// <https://html.spec.whatwg.org/multipage/#dom-messageevent-ports>
     fn Ports(&self, cx: JSContext) -> JSVal {
-        message_ports_to_frozen_array(self.ports.borrow().as_slice(), cx)
+        let ports: Vec<DomRoot<MessagePort>> = self
+            .ports
+            .borrow()
+            .iter()
+            .map(|port| DomRoot::from_ref(&**port))
+            .collect();
+        message_ports_to_frozen_array(ports.as_slice(), cx)
```

Comment:
We should update `message_ports_to_frozen_array` to work on `&[T] where T: ToJSValConvertible`  instead, and once we do that we should make it generic. But perhaps not in this PR.